geopandas/io/arrow.py

```diff
@@ -725,5 +781,120 @@ def _read_feather(path, columns=None, **kwargs):
         raise ImportError("pyarrow >= 0.17 required for Feather support")
 
     path = _expand_user(path)
+
     table = feather.read_table(path, columns=columns, **kwargs)
+    _validate_metadata(table.schema.metadata)
     return _arrow_to_geopandas(table)
+
+
+def _read_parquet_schema_and_metadata(path, filesystem):
+    ds = import_optional_dependency(
+        "pyarrow.dataset", extra="pyarrow is required for Feather support."
+    )
+    parquet = import_optional_dependency(
+        "pyarrow.parquet", extra="pyarrow is required for Parquet support."
+    )
+    try:
+        schema = ds.dataset(path, filesystem=filesystem).schema
+    except Exception:
+        schema = parquet.read_schema(path, filesystem=filesystem)
+
+    metadata = schema.metadata
+
+    # if GDAL-produced malformed file
+    if metadata is None or b"geo" not in metadata:
+        try:
+            # read_metadata does not accept a filesystem keyword, so need to
+            # handle this manually (https://issues.apache.org/jira/browse/ARROW-16719)
+            if filesystem is not None:
+                pa_filesystem = _ensure_arrow_fs(filesystem)
+                with pa_filesystem.open_input_file(path) as source:
+                    metadata = parquet.read_metadata(source).metadata
+            else:
+                metadata = parquet.read_metadata(path).metadata
+        except Exception:
+            pass
+
+    return schema, metadata
+
+
+def _validate_metadata(metadata):
+    if metadata is None or b"geo" not in metadata:
+        raise ValueError(
+            """Missing geo metadata in Parquet/Feather file.
+            Use pandas.read_parquet/read_feather() instead."""
+        )
+
+    # check for malformed metadata
+    try:
+        decoded_geo_metadata = _decode_metadata(metadata.get(b"geo", b""))
+    except (TypeError, json.decoder.JSONDecodeError):
+        raise ValueError("Missing or malformed geo metadata in Parquet/Feather file")
+
+    _validate_geo_metadata(decoded_geo_metadata)
+    return decoded_geo_metadata
+
+
+def _get_parquet_bbox_filter(geo_metadata, bbox):
+
+    _validate_bbox_column_in_parquet(geo_metadata)
+
+    bbox_column_name = _get_bbox_encoding_column_name(geo_metadata)
+    return _convert_bbox_to_parquet_filter(bbox, bbox_column_name)
+
+
+def _convert_bbox_to_parquet_filter(bbox, bbox_column_name):
+    import pyarrow
+    import pyarrow.compute as pc
+
+    if Version(pyarrow.__version__) < Version("9.0.0"):
+        raise ImportError(
+            "pyarrow >= 9.0.0 required to allow boolean expression to filter by bbox."
+        )
+
+    return (
+        (pc.field((bbox_column_name, "xmin")) >= bbox[0])
+        & (pc.field((bbox_column_name, "ymin")) >= bbox[1])
+        & (pc.field((bbox_column_name, "xmax")) <= bbox[2])
+        & (pc.field((bbox_column_name, "ymax")) <= bbox[3])
```

Comment:
I just realized (by testing this with the nz-building-outlines dataset, and doing a filtered read and comparing with gdal) that the above formula is not correct .. 
This works for geometries that are entirely within the bbox, but the expectation is that we get all geometries that are at least partly inside the bbox, i.e. all geometries that intersect with the bbox. 

For intersection, the formula is a bit different as for fully contained. I suppose negating that it is not outside as done in https://stackoverflow.com/questions/2752349/fast-rectangle-to-rectangle-intersection might be the easiest.